File: development/ui/src/app/map/domain/selected-feature-holder.ts
import {SelectedFeature} from "./selected-feature";

export class SelectedFeatureHolder {

  constructor(public listener: (SelectedFeature) => void) {
  }

  select(selection: SelectedFeature) {
    if (this.listener) {
      this.listener(selection);
    }
  }
}

File: production/client/src/main/scala/kpn/client/common/map/vector/SelectedFeatureHolder.scala
// TODO migrate to Angular
package kpn.client.common.map.vector

class SelectedFeatureHolder {

  var listener: Option[(Option[SelectedFeature]) => Unit] = None

  def select(selection: Option[SelectedFeature]): Unit = {
    listener.foreach(f => f(selection))
  }
}
